ariadne/wsgi_middleware.py

```diff
@@ -169,9 +182,8 @@ def return_response_from_result(
         status = HTTP_STATUS_200_OK
         response = {}
         if result.errors:
-            response["errors"] = [format_error(e) for e in result.errors]
-        if result.invalid:
             status = HTTP_STATUS_400_BAD_REQUEST
```

Comment:
Do we want to return 400 each time an error occurs? `apollo-server` will only return 400 if an error is thrown while parsing the query and/or variables. In such cases the response contains just the `errors` field and not `data`. In all other cases the status code is 200 and the `data` field is present (even if set to `null`).

Reply:
No opinion here. Originally we had nice `invalid` flag on the result we could've used to make the call. Now that it's gone, we could return 400 for errors coming from `validate_query`, and always return 200 when response is coming from `return_response_from_result`.

There's a gotcha here is that we can't reliably check `operationName` without looking into the parsed query's AST, so passing operation name thats string, but doesn't exist in the query could produce 200 with `"data": "null"`, but `errors` containing the message.

Reply:
I've changed this so we return 400 only for `validate_query` errors.